Gordon360/Controllers/ProfilesController.cs

```diff
@@ -476,6 +497,44 @@ public async Task<ActionResult<FacultyStaffProfileViewModel>> UpdateOfficeHours(
             return Ok(result);
         }
 
+        /// <summary>
+        /// Set visibility of some piece of personal data for user.
+        /// </summary>
+        /// <param name="userPrivacy">Faculty Staff Privacy Decisions (see UserPrivacyViewModel)</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("user_privacy")]
+        // [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.PROFILE_PRIVACY)]
+        public async Task<ActionResult<UserPrivacyViewModel>> UpdateUserPrivacyAsync(UserPrivacyViewModel userPrivacy)
+        {
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+            // The privacy setting of some fields (such as HomeCountry and Country) are controlled by one button,
+            // so they are conbined and sent by one API request, and we need to split them here.
+            string[] subFields = userPrivacy.Field.Split(' ');
+            foreach (var subField in subFields)
+            {
+                userPrivacy.Field = subField;
+                await _profileService.UpdateUserPrivacyAsync(authenticatedUserUsername, userPrivacy);
+            }
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// Return a list buildings.
+        /// </summary>
+        /// <returns> All buildings</returns>
+        [HttpGet]
+        [Route("visibility_group")]
+        public ActionResult<IEnumerable<string>> GetVisibilityGroup()
+        {
+            var groups = _context.UserPrivacy_Visibility_Groups.Select(up_v_g => up_v_g.Group)
+                                   .Distinct()
+                                   .Where(g => g != null)
+                                   .OrderBy(g => g);
```

Comment:
Why are we querying the database from within the controller? This should be done within a service function as per our API's design.

Reply:
The reason I'm writing this in the controller is because I've found that other people write it in the controller instead of in the service haha.. This is what I found in `AdvancedSearchController.cs`:
![image](https://github.com/gordon-cs/gordon-360-api/assets/78650688/3aa8b5cc-1895-4a67-b826-b1a94270863f)